base/logging.jl

```diff
@@ -290,7 +290,7 @@ function logmsg_code(_module, file, line, level, message, exs...)
     end
     # Note that it may be necessary to set `id` and `group` manually during bootstrap
     id !== nothing || (id = :(log_record_id(_module, level, $exs)))
-    group !== nothing || (group = :(Symbol(splitext(basename($file))[1])))
+    group !== nothing || (group = :(Symbol(splitext(basename(something($file, "")))[1])))
```

Comment:
For what it's worth, I think you can simplify this and the line above a bit by using:

```
id = something(id, :(log_record_id(_module, level, $exs)))
```